src/nni_manager/core/nniDataStore.ts

```diff
@@ -358,35 +339,94 @@ class NNIDataStore implements DataStore {
                     }
                 default:
             }
-            /* eslint-enable no-fallthrough */
             jobInfo.status = this.getJobStatusByLatestEvent(jobInfo.status, record.event);
+
             if (record.data !== undefined && record.data.trim().length > 0) {
-                const newHParam: any = this.parseHyperParameter(record.data);
-                if (newHParam !== undefined) {
-                    if (jobInfo.hyperParameters !== undefined) {
-                        let hParamIds: Set<number> | undefined = hParamIdMap.get(jobInfo.id);
-                        if (hParamIds === undefined) {
-                            hParamIds = new Set();
-                        }
-                        if (!hParamIds.has(newHParam.parameter_index)) {
-                            jobInfo.hyperParameters.push(JSON.stringify(newHParam));
-                            hParamIds.add(newHParam.parameter_index);
-                            hParamIdMap.set(jobInfo.id, hParamIds);
+                const hyperParameter: any = this.parseHyperParameter(record.data);
+                if (hyperParameter !== undefined) {
+                    // update trials here.
+                    const trialId = this.getTrialId(jobInfo.id, hyperParameter);
+                    let trialInfo: TrialJobInfo | undefined;
+                    if (trialMap.has(trialId)) {
+                        trialInfo = trialMap.get(trialId);
+                        if (trialInfo && trialInfo.hyperParameters) {
+                            trialInfo.hyperParameters.push(JSON.stringify(hyperParameter));
                         }
                     } else {
-                        assert(false, 'jobInfo.hyperParameters is undefined');
+                        trialInfo = {
+                            id: trialId,
+                            status: this.getJobStatusByLatestEvent('RUNNING', record.event),
+                            hyperParameters: [JSON.stringify(hyperParameter)],
+                            startTime: record.timestamp,
+                            sequenceId: sequenceId++
```

Comment:
`sequenceId` is generated by this function not from the record?

Reply:
It's not sequenceId of metric record, it's trial number shows on UI. It has value for first request only, for following are empty. so it needs to reassign value here. I changed the variable name to trialNumber for clearer.